orca-clouddriver/src/test/groovy/com/netflix/spinnaker/orca/kato/pipeline/QuickPatchStageSpec.groovy

```diff
@@ -16,302 +16,125 @@
-        "application/json",
-        objectMapper.writeValueAsBytes(["version": "1.2"])
-      )
-    )
-    when:
-    quickPatchStage.buildSteps(stage)
-
-    then:
-    1 * oortHelper.getInstancesForCluster(config, null, true, false) >> expectedInstances
-
-    and:
-    stage.context.skippedInstances.'i-2345'
-    stage.afterStages.size() == 1
-    with(stage.afterStages[0].context) {
-      application == application
-      account == account
-      region == region
-      clusterName == config.clusterName
-      instanceIds == ["i-1234"]
-      instances.size() == 1
-      instances.every {
-        it.value.hostName == expectedInstances.get(it.key).hostName
-        it.value.healthCheck == expectedInstances.get(it.key).healthCheck
-      }
-    }
-    where:
-    application = "deck"
-    region = "us-east-1"
-    account = "account"
-    asgNames = ["deck-prestaging-v300"]
-    instance1 = [instanceId: "i-1234", publicDnsName: "foo.com", health: [[foo: "bar"], [healthCheckUrl: "http://foo.com:7001/healthCheck"]]]
-    instance2 = [instanceId: "i-2345", publicDnsName: "foo2.com", health: [[foo2: "bar"], [healthCheckUrl: "http://foo2.com:7001/healthCheck"]]]
-    expectedInstances = ["i-1234": [hostName: "foo.com", healthCheckUrl: "http://foo.com:7001/healthCheck"], "i-2345": [hostName: "foo2.com", healthCheckUrl: "http://foo.com:7001/healthCheck"]]
   }
 
-  def "skipUpToDate with getVersion retries"() {
-    given:
-    def config = [
-      application : application,
-      clusterName : "deck-cluster",
-      account     : account,
-      region      : region,
-      baseOs      : "ubuntu",
-      skipUpToDate: true,
-      patchVersion: "1.2",
-      package     : "deck"
+  private Map mkInstance(int id) {
+    [
+      instanceId: "i-$id".toString(),
+      hostName: "h${id}.foo.com",
+      healthCheckUrl: "/health"
     ]
-    def stage = new PipelineStage(null, "quickPatch", config)
-    1 * quickPatchStage.createInstanceService(_) >> instanceService
-    4 * instanceService.getCurrentVersion(_) >> { throw new RetrofitError(null, null, null, null, null, null, null) }
-    1 * instanceService.getCurrentVersion(_) >>> new Response(
-      "foo", 200, "ok", [],
-      new TypedByteArray(
-        "application/json",
-        objectMapper.writeValueAsBytes(["version": "1.21"])
-      )
-    )
-
-    when:
-    quickPatchStage.buildSteps(stage)
-
-    then:
-    1 * oortHelper.getInstancesForCluster(config, null, true, false) >> expectedInstances
+  }
 
-    where:
-    application = "deck"
-    region = "us-east-1"
-    account = "account"
-    asgNames = ["deck-prestaging-v300"]
-    expectedInstances = ["i-1234": [hostName: "foo.com", healthCheckUrl: "http://foo.com:7001/healthCheck"]]
+  /**
+   * This noops out the spring batch aspects of stage building which aren't really a concern for the purposes
+   * of this test
+   */
+  @InheritConstructors
+  static class NoBatchQuickPatchStage extends QuickPatchStage {
```

Comment:
I could do this with a Spy - I tore this test apart because it was too much Mock magic but this last bit makes sense to keep the test simple